core/services/pg/q.go

```diff
@@ -154,6 +154,7 @@ func (q Q) WithOpts(qopts ...QOpt) Q {
 
 func (q Q) Context() (context.Context, context.CancelFunc) {
 	ctx := q.ParentCtx
+	// TODO: move parent context nil check to constructor to avoid nil error for queries that don't call Context()
```

Comment:
For this PR? If not, should we open and link to an issue?

Reply:
Not for this PR but wanted to discuss this. Going to add a PR description and some comments to discuss this right now

Reply:
Yeah, I want to discuss this before creating a ticket. But yeah, what is the purpose of Q.ParentCtx being nil (is there a use case for that). If not, can we initialize Q.ParentCtx to context.Background in the constructor? Reason being that you might want to access Q.ParentCtx for a query (ex: any Queryer method that needs a context passed in)

Reply:
That makes sense to me, but @samsondav is the expert.

Reply:
Jordan apparently you wrote this code (although you may have just moved it). I can't see how it can be `nil` actually probably its safe to remove this check.

Reply:
Yeah I'd say its safe to set the ParentCtx to context.Background() in the constructor and then remove this check